src/packages/emmett-mongodb/src/eventStore/mongoDBEventStore.ts

```diff
@@ -82,20 +86,49 @@ export type MongoDBReadEvent<EventType extends Event = Event> = ReadEvent<
 >;
 
 export type MongoDBEventStoreOptions = {
-  collection: Collection<EventStream>;
+  database?: string;
+  collection?: string;
   projections?: ProjectionRegistration<
     'inline',
     MongoDBReadEventMetadata,
     MongoDBProjectionInlineHandlerContext
   >[];
+} & (
+  | {
+      client: MongoClient;
+    }
+  | {
+      connectionString: string;
+      clientOptions?: MongoClientOptions;
+    }
+);
+
+export type MongoDBEventStore = EventStore<MongoDBReadEventMetadata> & {
+  close: () => Promise<void>;
 };
 
-export class MongoDBEventStore implements EventStore<MongoDBReadEventMetadata> {
-  private readonly collection: Collection<EventStream>;
+class MongoDBEventStoreImplementation implements MongoDBEventStore {
+  private readonly client: MongoClient;
+  private readonly defaultOptions: {
+    database: string | undefined;
+    collection: string | undefined;
```

Comment:
I think we should add some explicit documentation that not providing a `collection` would have each stream type separated into a separate coll., but providing a `collection` would _only_ use that for everything. Just to avoid it not being obvious that this changes how the streams are stored.

Maybe we also allow for a `collectionFor` option to allow the dev the option to deal with this.

Reply:
> I think we should add some explicit documentation that not providing a collection would have each stream type separated into a separate coll., but providing a collection would only use that for everything. Just to avoid it not being obvious that this changes how the streams are stored.

Yes, that's a good call, what would you say if we force user to make an explicit choice through the option and have event store options that'd have a field in event store option `{ storage: 'SINGLE_COLLECTION' }` or `{ storage: 'COLLECTION_PER_STREAM_TYPE' }` , that'd allow to discriminate user choice and force to e.g. provide collection for the first option, and client for the second (and having that mutually exclusive).

> Maybe we also allow for a collectionFor option to allow the dev the option to deal with this.

Just to double-check: would you give user an option to provide their own resolution function through store options? I like the idea!

Reply:
Yeah choosing single/multi collection storage I think is the way to go. I think something like this:

```
type Options =
    | { type: 'SINGLE_COLLECTION', collection?: string }
    | { type: 'COLLECTION_PER_STREAM_TYPE', collectionFor?: (streamType: string) => Promise<Collection> | Collection }
```

I think it can be the same type def as the private `collectionFor`. Could allow for grouping certain stream types together into the same collection, not that it's necessarily useful.

Reply:
I added issue for that: https://github.com/event-driven-io/emmett/issues/142